src/handlers/subjects/teacher.ts

```diff
@@ -0,0 +1,40 @@
+// GET
+import { CustomDynamoDB } from "../../dynamodb/database";
+import responseHelper from "../../helpers/responseHelper";
+import { validateToken } from "../../helpers/validations";
+
+const subjectsDB = new CustomDynamoDB('local-bootcamp-classroom-api-subjects', 'id')
+
+  
+export const handler = async(event: any) => {
+    const tokenData = await validateToken(event.headers.Authorization)
+
+    if(!tokenData){
+        return responseHelper(403, 'Fail Authenticating');
+    }
+
+    if(!event.pathParameters || event.pathParameters.teacherEmail.trim() === '' || tokenData.userType !== "TEACHER"){
+        return responseHelper(400, 'Teacher email is required')
+    }
+
+    const teacherEmail = event.pathParameters.teacherEmail;
+
+    const subject = await subjectsDB.query(teacherEmail, null, null, "teacherEmail-Index")
+
+    if(!subject || subject.length === 0){
+        return responseHelper(400, 'No subjects found')
+    }
+    
+    const subjectList = []
+
+    subject.forEach((item: any) => {
```

Comment:
This is a good approach, but a better one is to make a map function. It is almost the same as what you have done but uses less computational power
``` 
const mappedSubjects = subject.map((item: any) => {
    return {
        subjectId: item.id,
        .
        .
        .

    }
}
})